bin/git-fork

```diff
@@ -10,42 +10,77 @@ test -z "$url" && url=$(git remote get-url origin 2> /dev/null) && origin=true
 # validate repo url
 test -z "$url" && abort "github repo needs to be specified as an argument"
 
-# validate user
-echo "Enter your github username"
-read user
-[ -n "$user" ] || abort "git username required"
-echo "Enter github two-factor authentication code (leave blank if not set up)"
-read MFA_CODE
-
 # extract owner + project from repo url
 project=${url##*/}
 owner=${url%/$project}
 project=${project%.git}
-if [[ $owner == git@* ]]; then
-  owner=${owner##*:}
-else
-  owner=${owner##*/}
-fi
+owner=${owner##*[/:]}
+# Yes, the following extracts server for both SSH and https references
+server=${url##*@}
+server=${server##*://}
+server=${server%%[/:]*}
 
 # validate
 [ -z "$project" -o -z "$owner" ] && abort "github repo needs to be specified as an argument"
 
+# determine github credentials
+user=$(get_config_value "$server.user")
+token=$(get_config_value "$server.token")
+if [[ $(get_config_value "$server.add-api") == "true" ]]; then
+  api_server="api.$server"
+else
+  api_server=$server
+fi
+# retrieve the API prefix and clean it up (i.e. "api/v3/")
+api_prefix="$(get_config_value "$server.api-prefix")/"
+api_prefix=${api_prefix#/}
+api_prefix=${api_prefix/%\/\//\/}
+
+if [[ -z "$user" ]]; then
+  # validate user
+  echo "Enter your github username"
+  read user
+  [ -n "$user" ] || abort "git username required"
+fi
+
+if [[ -z "$token" ]]; then
+  echo "Enter github password"
+  read password
```

Comment:
@hickey 
I think the order is not critical. Just because let curl ask for your password is simpler.